sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/sonar/VideoTrackCheck.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SonarHTML :: SonarQube Plugin
+ * Copyright (c) 2010-2018 SonarSource SA and Matthijs Galesloot
+ * sonarqube@googlegroups.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sonar.plugins.html.checks.sonar;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.html.checks.AbstractPageCheck;
+import org.sonar.plugins.html.node.Attribute;
+import org.sonar.plugins.html.node.TagNode;
+
+@Rule(key = "S4084")
+public class VideoTrackCheck extends AbstractPageCheck {
+
+  @Override
+  public void startElement(TagNode node) {
+    if (isVideoTag(node) && hasVideoSrc(node) && !hasTrackChild(node)) {
+      createViolation(node.getStartLinePosition(), "Add subtitle files for this video.");
+    }
+  }
+
+  private static boolean hasVideoSrc(TagNode node) {
+    return node.getAttributes().stream().anyMatch(VideoTrackCheck::isSrcAttribute) ||
+      node.getChildren().stream().anyMatch(VideoTrackCheck::isSourceTag);
+  }
+
+  private static boolean hasTrackChild(TagNode node) {
+    return node.getChildren().stream().anyMatch(VideoTrackCheck::isTrackTag);
+  }
+
+  private static boolean isVideoTag(TagNode node) {
+    return "VIDEO".equalsIgnoreCase(node.getNodeName());
```

Comment:
please use node.equalsElementName which will cover the usage of local name as well.